apps/android/app/src/main/java/chat/simplex/app/views/TerminalView.kt

```diff
@@ -70,15 +81,18 @@ fun TerminalView(chatModel: ChatModel, close: () -> Unit) {
   }
 }
 
-private fun runAuth(authorized: MutableState<Boolean>, context: Context) {
+private fun authorizedPreviously(lastSuccessfulAuth: State<Long?>): Boolean =
+  lastSuccessfulAuth.value != null && SystemClock.elapsedRealtime() - lastSuccessfulAuth.value!! < 30 * 1e+3
+
+private fun runAuth(lastSuccessfulAuth: MutableState<Long?>, context: Context) {
   authenticate(
     generalGetString(R.string.auth_open_chat_console),
     generalGetString(R.string.auth_log_in_using_credential),
     context as FragmentActivity,
     completed = { laResult ->
       when (laResult) {
-        LAResult.Success, LAResult.Unavailable -> authorized.value = true
-        is LAResult.Error, LAResult.Failed -> authorized.value = false
+        LAResult.Success, LAResult.Unavailable -> lastSuccessfulAuth.value = SystemClock.elapsedRealtime()
+        is LAResult.Error, LAResult.Failed -> lastSuccessfulAuth.value = null
```

Comment:
```suggestion
        is LAResult.Success, LAResult.Unavailable -> lastSuccessfulAuth.value = SystemClock.elapsedRealtime()
        is LAResult.Error, LAResult.Failed -> lastSuccessfulAuth.value = null
```

also, lift out `lastSuccessfulAuth.value =` ?